### PR TITLE
linuxPackages.acpi-call: switch to nix-community fork

### PR DESCRIPTION
**Pull request by Mic92**
This fixes also build against linux 5.6
We also took the opportunity to cleanup the build.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mic92** [comment]
Result of `nixpkgs-review pr 84597` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
<br>- linuxPackages_testing_bcachefs.acpi_call
</details>
<details>
  <summary>16 packages built:</summary>
<br>- linuxPackages-libre.acpi_call
<br>- linuxPackages.acpi_call
<br>- linuxPackages_4_14.acpi_call
<br>- linuxPackages_4_19.acpi_call
<br>- linuxPackages_4_4.acpi_call
<br>- linuxPackages_4_9.acpi_call
<br>- linuxPackages_5_5.acpi_call
<br>- linuxPackages_5_6.acpi_call
<br>- linuxPackages_hardened.acpi_call
<br>- linuxPackages_latest-libre.acpi_call
<br>- linuxPackages_latest_hardened.acpi_call
<br>- linuxPackages_latest_xen_dom0.acpi_call
<br>- linuxPackages_latest_xen_dom0_hardened.acpi_call
<br>- linuxPackages_testing_hardened.acpi_call
<br>- linuxPackages_xen_dom0.acpi_call
<br>- linuxPackages_xen_dom0_hardened.acpi_call
</details>

File: pkgs/os-specific/linux/acpi-call/default.nix [hunk 1]
```diff
@@ -1,43 +1,31 @@
...
 stdenv.mkDerivation {
-  name = "acpi-call-${kernel.version}";
+  name = "acpi-call-2020-04-07-${kernel.version}";
 
-  src = fetchgit {
-    url = "git://github.com/mkottman/acpi_call.git";
-    rev = "ac67445bc75ec4fcf46ceb195fb84d74ad350d51";
-    sha256 = "0jl19irz9x9pxab2qp4z8c3jijv2m30zhmnzi6ygbrisqqlg4c75";
+  src = fetchFromGitHub {
+    owner = "Mic92";
```
Comment (7c6f434c) [inline pkgs/os-specific/linux/acpi-call/default.nix:7]: PR description says «nix-community», is this a before-transfer version?

Comment (Mic92) [reply pkgs/os-specific/linux/acpi-call/default.nix:7]: Good catch. Let me fix it!

File: pkgs/os-specific/linux/acpi-call/default.nix [hunk 2]
```diff
@@ -1,43 +1,32 @@
-{ stdenv, fetchgit, fetchpatch, kernel }:
+{ stdenv, fetchFromGitHub, fetchpatch, kernel }:
 
 stdenv.mkDerivation {
```
Comment (7c6f434c) [inline pkgs/os-specific/linux/acpi-call/default.nix:3]: need `rec` here if `src` is used in `meta`

File: pkgs/os-specific/linux/acpi-call/default.nix [hunk 3]
```diff
@@ -1,43 +1,32 @@
-{ stdenv, fetchgit, fetchpatch, kernel }:
+{ stdenv, fetchFromGitHub, fetchpatch, kernel }:
```
Comment (LEXUGE) [inline pkgs/os-specific/linux/acpi-call/default.nix:1]: Seems like `fetchpatch` is unused.

File: pkgs/os-specific/linux/acpi-call/default.nix [hunk 4]
```diff
@@ -1,43 +1,32 @@
-{ stdenv, fetchgit, fetchpatch, kernel }:
+{ stdenv, fetchFromGitHub, kernel }:
 
-stdenv.mkDerivation {
-  name = "acpi-call-${kernel.version}";
+stdenv.mkDerivation rec {
+  name = "acpi-call-2020-04-07-${kernel.version}";
```
Comment (LEXUGE) [inline pkgs/os-specific/linux/acpi-call/default.nix:4]: Why should we change the name of it?

Comment (Mic92) [reply pkgs/os-specific/linux/acpi-call/default.nix:4]: This is date of commit. It is part of the version number.

Comment (Mic92) [reply pkgs/os-specific/linux/acpi-call/default.nix]: I might make a new release instead soon though.

**Review by LEXUGE — APPROVED** [review]
That looks better.
LGTM

**Pull request merged**